src/main/java/org/jenkinsci/plugins/codescene/CodeSceneBuilder.java

```diff
@@ -324,6 +345,9 @@ public void perform(Run<?, ?> build, FilePath workspace, Launcher launcher, Task
             URL url = new URL(deltaAnalysisUrl);
 
             EnvVars env = build.getEnvironment(listener);
+            if(build instanceof WorkflowRun){
+                env.putAll(this.getEnvVar(((WorkflowRun)build).getExecution()));
```

Comment:
Can there be a potential problem with overriding the same var that already exists in `env`? Are all the env vars for pipeline jobs set only in the StepExecution-s env?

Reply:
in case your jenkins setup use pipelines the env vars could be get from workflow execution step